node/node_syncing.go

```diff
@@ -567,6 +595,60 @@ func (node *Node) getEncodedBlockByHash(hash common.Hash) ([]byte, error) {
 	return b, nil
 }
 
+func (node *Node) getEncodedBlockWithSigByHash(hash common.Hash) ([]byte, error) {
+	if b, ok := blockWithSigReqCache.Get(hash); ok {
+		return b.([]byte), nil
+	}
+	blk := node.Blockchain().GetBlockByHash(hash)
+	if blk == nil {
+		return nil, errBlockNotExist
+	}
+	sab, err := node.getCommitSigAndBitmap(blk)
```

Comment:
Should prefer the commitSig from nextBlock.CommitSig/Bitmap.  Only the last block in the chain rely on the commitSig in DB because it's not saved in the new block yet.

Reply:
Yes, this is exactly how currently the logic works. Please take a look into the function `getCommitSigAndBitmap`. 

Also to make the function name more clear, I renamed the function name `getCommitSigAndBitmapFromChildOrDB.`